rfc-124-deploying-banners.md

```diff
@@ -0,0 +1,81 @@
+# Use Edge Side Include (ESI) to deploy banners on GOV.UK
+
+## Summary
+
+Using Edge Side Include (ESI) to deploy banners on GOV.UK will help us move away from `static`/`slimmer` while being able to roll out changes quickly.
+
+## Background
+A few types of site-wide banners can be activated to convey important information on GOV.UK.
```

Comment:
Do we need to mention the survey banner here too? It's not something we tend to toggle on/off in the same way, but I assume we probably need to consider it here too.

Reply:
It's probably somehow similar to the cookie banner, in a sense that we anticipate updates instead of toggling, but it's a good point – I missed it. I'll try and look into it and include it as well.

Reply:
Correct me if I'm wrong, @vanitabarrett, but as far as I can see the survey banner is even more different than the other ones in terms of where the markup sits – [being exclusively generated by JavaScript](https://github.com/alphagov/static/blob/master/app/assets/javascripts/surveys.js#L15-L21).

Reply:
@alex-ju That's true, I hadn't realised it worked in that way. But I assume it will still need a central place to live if we're retiring Static?

Reply:
@vanitabarrett yes. I had a more in-depth look and, with minimal changes, it could be componentised 'as is', then potentially brought more in-line with the other banners and components in general through a few iterations. Since that's outside of the scope for this RFC I only added a commit to cover it in the background section. Thank you again for reminding me about it.